### DISCUSSION TITLE
How to add import path of a QML module defined in cxx-qt for qmllint?

### DISCUSSION THREAD

**Discussion by klochowicz** [KDAB/cxx-qt#974] [Q&A]
Hi! 

I'm trying to use `qmllint` for my QML application, but it cannot find the import path to the module built with cxx-qt.

I've noticed that deep down in the build directory there's `.qmltypes` etc that I suspect I need to import, but it's not exposed anywhere - would it be possible to somehow export QML type information into e.g. the same place where the generated c++ code is being placed?

**Comment by LeonMatthesKDAB** [comment]
Hi @klochowicz , you're right that this is currently not exported into a stable place.

We're currently reworking how we build our qml modules. We'll try to add the `.qmltypes` to a stable directory (e.g. the CXX_QT_EXPORT_DIR) so that they can be picked up by external tooling.

I hope that will add support for `qmllint` and `qmlls`

**Comment by iamsergio** [comment]
For `qmlls`, I've edited `build.rs` to generate a `.qmlls.ini` file in my source dir.

Something like:

```
fn main() {
    let out_dir = env::var("OUT_DIR").expect("OUT_DIR not set");
    let out_dir = format!("{}/qt-build-utils/qml_modules/", out_dir);
    let qmlls_ini_in =
        std::fs::read_to_string("qmlls.ini.in").expect("Failed to read template qmlls.ini.in");

    let qmlls_ini = qmlls_ini_in
        .replace("@CXXQT_OUT_DIR@", &out_dir);

    std::fs::write(".qmlls.ini", qmlls_ini).expect("Failed to write to .qmlls.ini");

```

For qmllint you could generate shell script, as a workaround.

CC #975

**Reply by klochowicz** [reply]
very promising, I'll try that out!
